### ISSUE TITLE
[HOLD #19348][$1000] Not showing deleted message immediately

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19901]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Open any report
3. Send any message
4. Click on reply to thread on message
5. Send any message on thread
6. Now, Click on delete and confirm deleting the main thread message
7. You will see message saying "[Deleted message]" is not shown immediately
8. Go back to LHN and again open chat where you have deleted main thread message on step 6
9. After some time the message you have deleted will change into "[Deleted message]"

## Expected Result:
The, "[Deleted message]" should be shown immediately after performing deletion action

## Actual Result:
The, "[Deleted message]" is not shown when, same room is revisited. Only updates after changing report and then return.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.20.4

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/2c8f5fef-7cc9-4d31-aefb-c297a662c391

https://github.com/Expensify/App/assets/93399543/ef8bbd0a-0b9f-459a-9cec-ae9adb3d57ec

**Expensify/Expensify Issue URL:**

**Issue reported by:** @abhilup

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684943161522479

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0140be6f5f1549e6e5</li>
        <li>Upwork Job ID: 1667196891808641024</li>
        <li>Last Price Increase: 2023-06-09</li>
    </ul>
</details>

**Comment by luacmartins** [comment]
@allroundexperts it seems like you pointed this issue having the same root cause as https://github.com/Expensify/App/issues/19348. Since that one is older and has more activity on, I'm gonna put this on hold until we fix that issue and then we should retest this issue.

**Comment by trjExpensify** [comment]
👋 seems like there are two bug reports from @abhilup co-mingled in this issue. Do we need to hold on #19348 to remove the delete button from the comment action menu on deleted messages? ([bug report](https://expensify.slack.com/archives/C049HHMV9SM/p1684412217524939))

**Comment by trjExpensify** [comment]
^^ bump. @luacmartins, is it necessary to hold removing the delete button from the comment action menu here?

**Comment by Nathan-Mulugeta** [comment]
Hey team, I came here from [this](https://expensify.slack.com/archives/C049HHMV9SM/p1684754620527369) slack discussion, I just want to let you know that if this open GH issue is related to my issue on slack I think my slack report came first which was on May 24th.

**Comment by trjExpensify** [comment]
Are you talking about the delete button on deleted messages portion yeah?

**Comment by trjExpensify** [comment]
Cool, that was reported here on May 18th: https://expensify.slack.com/archives/C049HHMV9SM/p1684412217524939

**Comment by Nathan-Mulugeta** [comment]
Okay I see, is the issue of the [Deleted message] re appearing after being deleted related to this issue?

**Comment by luacmartins** [comment]
@trjExpensify good point! We don't need to hold to remove the Delete button from the context menu. Removed hold.

**Comment by luacmartins** [comment]
Haha so it seems like we can remove the `[Delete message] can still be deleted` part of this issue and focus on `Not showing deleted message immediately`? So we're back on hold then?

**Comment by luacmartins** [comment]
I edited the OP to reflect that

**Comment by trjExpensify** [comment]
Haha, makes sense!

**Comment by luacmartins** [comment]
Still on hold

**Comment by abdulrahuman5196** [comment]
Still on hold

**Comment by luacmartins** [comment]
The holding issue has been closed. Can we retest this one please? @stephanieelliott @abdulrahuman5196

**Comment by abdulrahuman5196** [comment]
@luacmartins That hold issue was closed due to an internal effort to solve issues revolving around thread edits and reactions. 
https://github.com/Expensify/App/issues/19348#issuecomment-1593164126

I don't think this issue would have solved. I think this should also be tracked as part of the internal issue, but waiting on the confirmation from @chiragsalian  https://github.com/Expensify/App/issues/19901#issuecomment-1593168194.
If you are able to confirm and close this out will also be great.

**Comment by luacmartins** [comment]
>@chiragsalian I expect this to be solved as well pertaining to this https://github.com/Expensify/App/issues/19348#issuecomment-1593164126 since both has similar root cause.

@chiragsalian can you confirm the above?

**Comment by luacmartins** [comment]
I'll get back to this on monday

**Comment by chiragsalian** [comment]
Yeah i think we should close out this issue. 
It should be tackled by this [one](https://github.com/Expensify/Expensify/issues/282991).

**Comment by luacmartins** [comment]
Thanks for confirming. Closing!
